src/Twig/Extension/CanonicalizeExtension.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Twig\Extension;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class CanonicalizeExtension extends AbstractExtension
+{
+    // @todo: there are more locales which are not supported by moment and they need to be translated/normalized/canonicalized here
+    private const MOMENT_UNSUPPORTED_LOCALES = [
+        'de' => ['de', 'de-at'],
+        'es' => ['es', 'es-do'],
+        'nl' => ['nl', 'nl-be'],
+        'fr' => ['fr', 'fr-ca', 'fr-ch'],
+    ];
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    public function __construct(RequestStack $requestStack)
```

Comment:
```suggestion
    /**
     * @internal This class is only meant to use it through Twig
     */
    public function __construct(RequestStack $requestStack)
```
IMO we could mark these constructors as `@internal` since the methods should only be accessed through Twig.